### ISSUE TITLE
Tracking: Add a property for install / datasource events that specifies where it happened

### ISSUE THREAD

**Issue by leventebalogh** [grafana/grafana#53208]
We would like to see how our users interact with the new Data Connections page in the cloud, so we would like to introduce some user tracking for this, possibly by using RudderStack.

**Comment by tolzhabayev** [comment]
We would need to define areas that we want to track first, I would suggest @teddyba - maybe you can create a proposal together with @richardqlam on:
- what do we already track in the app?
- what is missing in to track in the app?
- what do we want to track on-prem?

**Comment by richardqlam** [comment]
@mikkancso @besartberisha What Rudderstack events are instrumented in the onboarding plugin today? I know today we're able to track an end to end workflow for an integration, and are able to see when a user selects an integration, switches between tabs on the details page, and clicks the install button and finishes installation.


Ideally, for the Connections page, we'd want to track:
- When a user lands on the page
- Which previous page the user came from
- What filters they clicked
- Whether they searched or not and what they searched
- Which integration they selected
- When they start installing a connection and whether they've successfully configured something

**Comment by mikkancso** [comment]
@richardqlam The current list of rudderstack events can be found [here](https://github.com/grafana/cloud-onboarding/blob/6606f904c324ea649d87b1ff7234fc85c10d5223/plugin/src/utils/enums.ts#L94-L106).

**Comment by mikkancso** [comment]
FTR, I just merged a PR which adds some events to our list:
```
  PageView = 'Page View',
  ViewDashboards = 'cloud_user_clicked_integration_view_dashboards_button',
  AbandonUninstall = 'cloud_user_clicked_abandon_uninstall_integration',
  InstallButton = 'cloud_user_clicked_install_integration_button',
  UninstallButton = 'cloud_user_clicked_uninstall_integration_button',
  UpdateButton = 'cloud_user_clicked_update_integration_button',
  RemoveIntegration = 'cloud_user_clicked_remove_integration_button',
  InstallSuccess = 'cloud_user_install_integration_succeeded',
  InstallFailed = 'cloud_user_install_integration_failed',
  BackButton = 'cloud_user_clicked_back_to_all_integrations_button',
  DetailsTabClick = 'cloud_user_clicked_integration_configuration_details_tab', 
+ CardClick = 'cloud_user_clicked_integration_card',
+ SearchInput = 'cloud_user_typed_input_in_search',
+ FilterClick = 'cloud_user_clicked_category_filter_pill',
+ FilterSetFromUrl = 'category_filter_set_from_url',
+ TestConnectionClick = 'cloud_user_clicked_test_connection',
+ TestConnectionSucceeded = 'test_connection_succeeded',
+ TestConnectionFailed = 'test_connection_failed',
```

So I think the following events are missing now:
- Which previous page the user came from
- All / Installed / New updates filters

We also have to think about this question:
- what do we want to track on-prem?

**Comment by mikkancso** [comment]
> what do we want to track on-prem?

I'm not sure if we can track anything on-prem, so I would rephrase this question:
What parts of the OSS Grafana do we want to track when it is running in Cloud?

And we recently talked about this: data source installs would be nice to track.

**Comment by mikkancso** [comment]
> And we recently talked about this: data source installs would be nice to track.

They are already tracked: https://github.com/grafana/grafana/blob/main/public/app/features/datasources/tracking.ts

**Comment by leventebalogh** [comment]
Nice @mikkancso! 👏 

I guess then we probably need to add a new property to these events which tells where (Connections or Administration) the datasource was created / tested. I will create an issue for that.
